opm/parser/eclipse/EclipseState/InitConfig/FoamConfig.hpp

```diff
@@ -23,44 +23,54 @@
 #include <cstddef>
 #include <vector>
 
-namespace Opm {
-
-    class Deck;
-    class DeckRecord;
-
-    class FoamRecord {
-        public:
-            explicit FoamRecord( const DeckRecord& );
-
-            double referenceSurfactantConcentration() const;
-            double exponent() const;
-            double minimumSurfactantConcentration() const;
-
-        private:
-            double reference_surfactant_concentration_;
-            double exponent_;
-            double minimum_surfactant_concentration_;
-    };
-
-    class FoamConfig {
-        public:
-            using const_iterator = std::vector< FoamRecord >::const_iterator;
-
-            FoamConfig() = default;
-            explicit FoamConfig( const Deck& );
-
-            const FoamRecord& getRecord( std::size_t index ) const;
-
-            std::size_t size() const;
-            bool empty() const;
-
-            const_iterator begin() const;
-            const_iterator end() const;
-
-        private:
-            std::vector< FoamRecord > records;
-    };
-
-}
+namespace Opm
+{
+
+class Deck;
+class DeckRecord;
+
+/// Foam behaviour data for a single SATNUM region.
+class FoamData
+{
+public:
+    explicit FoamData(const DeckRecord& FOAMFSC_record, const DeckRecord& FOAMROCK_record);
+
+    double referenceSurfactantConcentration() const;
+    double exponent() const;
+    double minimumSurfactantConcentration() const;
+
+    enum class FoamAllowDesorption { Yes = 1, No = 2 };
```

Comment:
That is a `bool` in my book; you could use: [this](https://github.com/OPM/opm-common/blob/f01ed130af0bae64c9a922b63a0de7977c5311e9/src/opm/parser/eclipse/Deck/DeckItem.cpp#L452) for conversion.

Reply:
Will change.